### DISCUSSION TITLE
Is there a way to override existing attributes, or compare if type A satisfies type B?

### DISCUSSION THREAD

**Discussion by zanzlender** [colinhacks/zod#3221] [Q&A]
Hello, I've recently come across a use case like this but could find an answer if something like this was possible, or exists. Let me explain in more detail.

I have been using Drizzle, with drizzle-zod package to generate my Zod type from the Table schemas, so from the schema I can generate a insert Zod type: 

```ts
export const users = createTable("user", {
  id: varchar("id", { length: 255 }).primaryKey()
  name: varchar("name", { length: 255 }),
  email: varchar("email", { length: 255 }),
});

export const createUserSchema = createInsertSchema(users);
```

This all works great 👍 However, since none of the defined attributes in the created schema have error messages I have to generate them myself.  So I take the generated schema and Pick those properties that I need for my specific use case. Let's say I need a schema to update the user's "Personal info" (simplified for the example). Then I Extend that type to add the messages I need like so.

```ts
export const zPersonalInfoSchema = createUserSchema
  .pick({
    id: true,
    name: true,
  })
  .extend({
    id: z
      .string({ required_error: "id required" }),
    name: z
      .string({ required_error: "Name is required" })
      .min(1, { message: "Name is required" })
      .max(100, { message: "Maximum 100 chars allowed" }),
  });

export type PersonalInfoSchema = z.infer<typeof zPersonalInfoSchema >;
```

However, the problem now is since I extend I can add whatever I want, not only the attributes that were already there. Is there a way to "Override" existing types, or if I were to create a new Zod type from scratch, is there a way to check if my new Zod type B "satisfies" the existing type A?

Some thing like this I would use only as a layer to catch potential bugs which may occur when changing the database schema... I think it would be useful, but tell me what you all think.

**Comment by jwnx** [comment]
I'm also interested in this use-case for [`kanel-zod`](https://github.com/kristiandupont/kanel).
